includes/polling/functions.inc.php

```diff
@@ -773,3 +774,41 @@ function data_flatten($array, $prefix = '', $joiner = '_')
 
     return $return;
 }
+
+/**
+ * @param string $sysLocation location override (instead of sysLocation.0)
+ * @param &$device
+ * @param &$update_array
+ */
+function set_device_location($sysLocation, &$device, &$update_array)
+{
+    $sysLocation = str_replace('"', '', $sysLocation);
+
+    // Rewrite sysLocation if there is a mapping array (database too?)
+    if (!empty($sysLocation) && (is_array(Config::get('location_map')) || is_array(Config::get('location_map_regex')) || is_array(Config::get('location_map_regex_sub')))) {
+        $sysLocation = rewrite_location($sysLocation);
+    }
+
+    if ($sysLocation == 'not set') {
```

Comment:
same here ...
Device specific default value?
shouldn't be handled in LibreNMS

Reply:
I believe that check was in polling/core.inc.php already. All I did here was move the syslocation code into a function, so it can also be called in a custom OS polling file.
Not sure where "not set" comes from

Reply:
mmh, strange,
ok let it in for now